src/Sulu/Bundle/MediaBundle/Controller/MediaStreamController.php

```diff
@@ -116,13 +119,35 @@ protected function getFileResponse(
         $fileSize = $fileVersion->getSize();
         $storageOptions = $fileVersion->getStorageOptions();
         $mimeType = $fileVersion->getMimeType();
-        $version = $fileVersion->getVersion();
+        $storageName = $fileVersion->getStorageName();
+        $pathInfo = pathinfo($fileName);
 
-        $path = $this->getStorage()->load($fileName, $version, $storageOptions);
+        $downloadUrl = $this->getStorageManager()->getDownloadUrl($storageOptions, $storageName);
 
-        $response = new BinaryFileResponse($path);
+        // If external download Url exists redirect to it.
+        if (!empty($downloadUrl)) {
+            return new RedirectResponse($downloadUrl);
+        }
 
-        $pathInfo = pathinfo($fileName);
+        $resourcePath = $this->getStorageManager()->load($storageOptions, $storageName);
+
+        if (is_resource($resourcePath)) {
+            $handle = $resourcePath;
```

Comment:
here we should find maybe a better solution for handling with resources.
maybe we can use `stream_get_meta_data` and access `uri` to use `BinaryFileResponse`.
It would be good so the localStorage could also return a `resource` instead of a `string`